src/Common/src/TypeSystem/TypesDebugInfoWriter/UserDefinedTypeDescriptor.cs

```diff
@@ -14,17 +14,27 @@ public UserDefinedTypeDescriptor(ITypesDebugInfoWriter objectWriter)
         {
             _objectWriter = objectWriter;
         }
-        public uint GetTypeIndex(TypeDesc type)
+        public uint GetTypeIndex(TypeDesc type, bool needsCompleteType)
         {
             uint variableTypeIndex = 0;
-            if (!_knownTypes.TryGetValue(type, out variableTypeIndex))
```

Comment:
Nit: This could be collapsed into a simpler expression.
```
uint variableTypeIndex = 0;
if (needsCompleteType ? _completeKnownTypes.TryGetValue(type, out variableTypeIndex) : _knownTypes.TryGetValue(type, out variableTypeIndex))
    return variableTypeIndex;

return GetNewTypeIndex(type, needsCompleteType);
```